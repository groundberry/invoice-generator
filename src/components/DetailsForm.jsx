import React, { Component } from "react";
import "./DetailsForm.css";

class DetailsForm extends Component {
  render() {
    const { value } = this.props;
    const { name, email, address, phone } = value;

    return (
      <form className="DetailsForm">
        <label>
          Name:
          <input
            type="text"
            name="name"
            value={name}
            onChange={this.handleChangeName}
          />
        </label>
        <label>
          Email:
          <input
            type="email"
            name="email"
            value={email}
            onChange={this.handleChangeEmail}
          />
        </label>
        <label>
          Address:
          <input
            type="text"
            name="address"
            value={address}
            onChange={this.handleChangeAddress}
          />
        </label>
        <label>
          Phone number:
          <input
            type="tel"
            name="phone"
            value={phone}
            onChange={this.handleChangePhone}
          />
        </label>
      </form>
    );
  }
  handleChangeName = evt => {
    const { value } = this.props;
    this.props.onChange({
      ...value,
      name: evt.target.value
    });
  };

  handleChangeEmail = evt => {
    const { value } = this.props;
    this.props.onChange({
      ...value,
      email: evt.target.value
    });
  };

  handleChangeAddress = evt => {
    const { value } = this.props;
    this.props.onChange({
      ...value,
      address: evt.target.value
    });
  };

  handleChangePhone = evt => {
    const { value } = this.props;
    this.props.onChange({
      ...value,
      phone: evt.target.value
    });
  };
}

export default DetailsForm;
