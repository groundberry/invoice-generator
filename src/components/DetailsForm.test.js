import React from "react";
import { shallow } from "enzyme";
import DetailsForm from "./DetailsForm";

const data = {
  name: "John",
  email: "john@doe.com",
  address: "1st Ave",
  phone: "555 666 7777"
};

describe("DetailsForm", () => {
  let wrapper;
  let onChange;

  beforeEach(() => {
    onChange = jest.fn();
    wrapper = shallow(<DetailsForm value={data} onChange={onChange} />);
  });

  it("renders", () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe("when name changes", () => {
    beforeEach(() => {
      wrapper
        .find('input[name="name"]')
        .simulate("change", { target: { value: "Jane" } });
    });

    it("invokes onChange with the new name", () => {
      expect(onChange).toHaveBeenCalledWith({ ...data, name: "Jane" });
    });
  });

  describe("when email changes", () => {
    beforeEach(() => {
      wrapper
        .find('input[name="email"]')
        .simulate("change", { target: { value: "jane@doe.com" } });
    });

    it("invokes onChange with the new email", () => {
      expect(onChange).toHaveBeenCalledWith({ ...data, email: "jane@doe.com" });
    });
  });

  describe("when address changes", () => {
    beforeEach(() => {
      wrapper
        .find('input[name="address"]')
        .simulate("change", { target: { value: "2nd Ave" } });
    });

    it("invokes onChange with the new address", () => {
      expect(onChange).toHaveBeenCalledWith({ ...data, address: "2nd Ave" });
    });
  });

  describe("when phone changes", () => {
    beforeEach(() => {
      wrapper
        .find('input[name="phone"]')
        .simulate("change", { target: { value: "555 666 8888" } });
    });

    it("invokes onChange with the new phone", () => {
      expect(onChange).toHaveBeenCalledWith({ ...data, phone: "555 666 8888" });
    });
  });
});
