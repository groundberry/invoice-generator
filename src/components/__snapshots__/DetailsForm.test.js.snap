// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsForm renders 1`] = `
<form
  className="DetailsForm"
>
  <label>
    Name:
    <input
      name="name"
      onChange={[Function]}
      type="text"
      value="John"
    />
  </label>
  <label>
    Email:
    <input
      name="email"
      onChange={[Function]}
      type="email"
      value="john@doe.com"
    />
  </label>
  <label>
    Address:
    <input
      name="address"
      onChange={[Function]}
      type="text"
      value="1st Ave"
    />
  </label>
  <label>
    Phone number:
    <input
      name="phone"
      onChange={[Function]}
      type="tel"
      value="555 666 7777"
    />
  </label>
</form>
`;
